specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2022-10-01/costmanagement.generatedetailedcostreport.json

```diff
@@ -0,0 +1,373 @@
+        "externalDocs": {
+          "url": "https://docs.microsoft.com/en-us/rest/api/costmanagement/"
+        },
+        "x-ms-examples": {
+          "GenerateDetailedCostReportByBillingProfileAndInvoiceId": {
+            "$ref": "./examples/GenerateDetailedCostReportByBillingProfileAndInvoiceId.json"
+          },
+          "GenerateDetailedCostReportBySubscriptionAndTimePeriod": {
+            "$ref": "./examples/GenerateDetailedCostReportBySubscriptionAndTimePeriod.json"
+          },
+          "GenerateDetailedCostReportByCustomerAndTimePeriod": {
+            "$ref": "./examples/GenerateDetailedCostReportByCustomerAndTimePeriod.json"
+          },
+          "GenerateDetailedCostReportByBillingProfileAndInvoiceIdAndCustomerId": {
+            "$ref": "./examples/GenerateDetailedCostReportByBillingProfileAndInvoiceIdAndCustomerId.json"
+          },
+          "GenerateDetailedCostReportByBillingAccountLegacyAndBillingPeriod": {
+            "$ref": "./examples/GenerateDetailedCostReportByBillingAccountLegacyAndBillingPeriod.json"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/scopeUsageDetailsParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/GenerateDetailedCostReportDefinition"
+            },
+            "description": "Parameters supplied to the Create detailed cost report operation."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request processing completed.",
+            "schema": {
+              "$ref": "#/definitions/GenerateDetailedCostReportOperationResult"
+            }
+          },
+          "202": {
+            "description": "Accepted. Request will be processed. Use the Location header to check the status.",
+            "headers": {
+              "Location": {
+                "description": "The URL to check the result of the asynchronous operation.",
+                "type": "string"
+              },
+              "Azure-Consumption-AsyncOperation": {
+                "description": "The URL to check the status of the asynchronous operation.",
+                "type": "string"
+              },
```

Comment:
I have a doubt about extending the async api this way. Wouldn't the final response of the POST operation already be defined/retrieved by following the 202 Location redirect unitl it returns 200? Why have this extra way to get the operation and the final operation result?

Reply:
This is an existing API. I missed adding this in the initial commit.

Adding @absyed to respond.

Reply:
Spoke with @absyed offline. These have been added for backward compatibility with a legacy API.